src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebeforecasedefaultcolon/InputNoWhitespaceBeforeCaseDefaultColon.java

```diff
@@ -0,0 +1,54 @@
+package com.puppycrawl.tools.checkstyle.checks.whitespace.nowhitespacebeforecasedefaultcolon;
+
+/*
+* Config = default
+*/
+public class InputNoWhitespaceBeforeCaseDefaultColon {
+    {
+        switch(1) {
+            case 1 : // violation
+                break;
+            case 2:
+                break;
+            default : // violation
+                break;
+        }
+
+        switch(2) {
+            case 2:
+                break;
+            default:
+                break;
+
+        }
+
+        switch(3) {
+            case 3 /* space around*/ : // violation
+                break;
+            default/* no space around */:
```

Comment:
to support this case when logic is based on AST only, you need to make Check to be comments aware, theres is special getter in base class that make Check receive all comments. 
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java#L85

Reply:
Done.